packages/firestore/test/unit/specs/existence_filter_spec.test.ts

```diff
@@ -286,4 +246,347 @@ describeSpec('Existence Filters:', [], () => {
+      const bloomFilterProto = generateBloomFilterProto({
+        contains: [docA],
+        notContains: [docB],
+        numOfBits: 2,
+        hashCount: 1
+      });
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA, docB)
+          .expectEvents(query1, { added: [docA, docB] })
+          // DocB is deleted in the next sync.
+          .watchFilters([query1], [docA.key], bloomFilterProto)
+          .watchSnapshots(2000)
+          // BloomFilter identify docB is deleted, skip full query and put docB
+          // into limbo directly.
+          .expectEvents(query1, { fromCache: true })
+          .expectLimboDocs(docB.key) // DocB is now in limbo.
+          .ackLimbo(2000, deletedDoc('collection/b', 2000))
+          .expectLimboDocs() // DocB is no longer in limbo.
+          .expectEvents(query1, {
+            removed: [docB]
+          })
+      );
+    }
+  );
+
+  specTest(
+    'Full re-query is triggered when bloom filter can not identify documents deleted',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { v: 1 });
+      const docB = doc('collection/b', 1000, { v: 2 });
+      const docC = doc('collection/c', 1000, { v: 2 });
+      const bloomFilterProto = generateBloomFilterProto({
+        contains: [docA, docB],
+        notContains: [docC],
+        numOfBits: 3,
+        hashCount: 1
+      });
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA, docB, docC)
+          .expectEvents(query1, { added: [docA, docB, docC] })
+          // DocB and DocC are deleted in the next sync.
+          .watchFilters([query1], [docA.key], bloomFilterProto)
+          .watchSnapshots(2000)
+          // BloomFilter correctly identifies docC is deleted, but yields false
+          // positive results for docB. Re-run query is triggered.
+          .expectEvents(query1, { fromCache: true })
+          .expectActiveTargets({ query: query1, resumeToken: '' })
+          .watchRemoves(query1) // Acks removal of query.
+          .watchAcksFull(query1, 2000, docA)
+          .expectLimboDocs(docB.key, docC.key) // DocB, docC are now in limbo.
+          .ackLimbo(2001, deletedDoc('collection/b', 2001))
+          .expectEvents(query1, {
+            removed: [docB],
+            fromCache: true
+          })
+          .expectLimboDocs(docC.key)
+          .ackLimbo(2002, deletedDoc('collection/c', 2002))
+          .expectEvents(query1, {
+            removed: [docC]
+          })
+          .expectLimboDocs()
+      );
+    }
+  );
+
+  specTest(
+    'Bloom filter can process special characters in document name',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/ÀÒ∑', 1000, { v: 1 });
+      const docB = doc('collection/À∑Ò', 1000, { v: 1 });
+      const bloomFilterProto = generateBloomFilterProto({
```

Comment:
To improve readability, could you still use generateBloomFilterProto() and then just modify the returned object to meet the criteria of the test?

For example, in this test could you do something like this:
```ts
const bloomFilterProto = generateBloomFilterProto({...});
delete bloomFilterProto.hashCount
delete bloomFilterProto.bits.padding
```

If this works, please do this for the rest of the tests that test for invalid input. This way the test is VERY clear about what it's testing.